hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -1131,6 +1196,25 @@ private static Schema getActualSchemaFromUnion(Schema schema, Object data) {
     return actualSchema;
   }
 
+  private static Schema getActualSchemaFromUnion(Schema schema) {
+    Schema actualSchema;
+    if (schema.getType() != UNION) {
+      return schema;
+    }
+    if (schema.getTypes().size() == 2
+        && schema.getTypes().get(0).getType() == Schema.Type.NULL) {
+      actualSchema = schema.getTypes().get(1);
+    } else if (schema.getTypes().size() == 2
+        && schema.getTypes().get(1).getType() == Schema.Type.NULL) {
+      actualSchema = schema.getTypes().get(0);
+    } else if (schema.getTypes().size() == 1) {
+      actualSchema = schema.getTypes().get(0);
```

Comment:
Is there any way to share this logic with the method above?

Reply:
+1